lib/mix/phoenix.ex

```diff
@@ -80,8 +80,8 @@ defmodule Mix.Phoenix do
       {k, {:array, _}} -> {k, []}
       {k, :belongs_to} -> {k, nil}
       {k, :integer}    -> {k, 42}
-      {k, :float}      -> {k, "120.5"}
-      {k, :decimal}    -> {k, "120.5"}
+      {k, :float}      -> {k, 120.5}
+      {k, :decimal}    -> {k, Decimal.new(43.20)}
```

Comment:
> Anyway, no error, but saw the sample data generated in the tests used the string "120.5" for decimal. Ran validate_number/3 for the decimal field and it would always pass with the string.

If this is the case... then we have a bug. The value should be cast up to the point we reach validate_number. However i have just realized validate_number likely won't work with Decimal. So we need to fix it anyway.

Other than that, I don't think we need to change anything in Phoenix. :)